extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/DevConsoleCORSFilter.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkus.vertx.http.runtime.devmode;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.logging.Logger;
+
+import io.quarkus.vertx.http.runtime.cors.CORSConfig;
+import io.quarkus.vertx.http.runtime.cors.CORSFilter;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+
+public class DevConsoleCORSFilter implements Handler<RoutingContext> {
+    private static final Logger LOG = Logger.getLogger(DevConsoleCORSFilter.class);
+
+    private static final String HTTP_PORT_CONFIG_PROP = "quarkus.http.port";
+    private static final String LOCAL_HOST = "localhost";
+    private static final String HTTP_LOCAL_HOST = "http://" + LOCAL_HOST;
+    private static final String HTTPS_LOCAL_HOST = "https://" + LOCAL_HOST;
+
+    public DevConsoleCORSFilter() {
+    }
+
+    private static CORSFilter corsFilter() {
+        int httpPort = ConfigProvider.getConfig().getValue(HTTP_PORT_CONFIG_PROP, int.class);
```

Comment:
I guess it does not really matter because its just the dev console, but you could have just injected the config at creation rather that doing all of this for every request.

Reply:
@stuartwdouglas, I can't make the injection work, we create this handler at build time manually, so I think this is why it does not work - not sure it is worth introducing a runtime init for this handler; If you have some ideas how to handle it neatly for devconsole handlers then let me know please